backend/src/main/java/ca/mcgill/purposeful/controller/ReactionController.java

```diff
@@ -0,0 +1,55 @@
+package ca.mcgill.purposeful.controller;
+
+import ca.mcgill.purposeful.dto.ReactionDTO;
+import ca.mcgill.purposeful.exception.GlobalException;
+import ca.mcgill.purposeful.model.Reaction;
+import ca.mcgill.purposeful.model.Reaction.ReactionType;
+import ca.mcgill.purposeful.service.ReactionService;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * API for accessing the endpoints of Reaction
+ */
+@RestController
+@RequestMapping({"/api/reaction", "/api/reaction/"})
+public class ReactionController {
+
+  @Autowired
+  private ReactionService reactionService;
+
+  /**
+   * POST method to react
+   *
+   * @param reactionDTO - the reaction added or removed
+   * @return the newly created reaction DTO or the removed reaction DTO with null values
+   * @author Athmane Benarous
+   */
+  @PostMapping(value = {"/react", "/react/"})
+  @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
```

Comment:
Shouldn't we authenticate the user reacting so that you cannot react for someone else? 
e.g. user1 reacts to an idea by passing the id of user2, thus making user2 react to the idea

Reply:
Done!